src/patient-registration/patient-registration.component.tsx

```diff
@@ -83,22 +90,99 @@ export const PatientRegistration: React.FC = () => {
+            initialFormValues.familyName = name.family;
+          }
+          if (index === 1) {
+            patientUuidMap['additionalNameUuid'] = name.id;
+            initialFormValues.addNameInLocalLanguage = true;
+            initialFormValues.additionalGivenName = name.given[0];
+            initialFormValues.additionalMiddleName = name.given[1];
+            initialFormValues.additionalFamilyName = name.family;
+          }
+        });
+      } else {
+        initialFormValues.unidentifiedPatient = true;
+      }
+      initialFormValues.gender = capitalize(existingPatient.gender);
+      initialFormValues.birthdate = existingPatient.birthDate;
+      initialFormValues.telephoneNumber = existingPatient.telecom ? existingPatient.telecom[0].value : '';
+
+      existingPatient.identifier.forEach(id => {
+        const key = id.system ? camelCase(id.system) : camelCase(id.type.text);
+        patientUuidMap[key] = {
+          uuid: id.id,
+          value: id.value,
+        };
+        initialFormValues[key] = id.value;
+      });
+
+      if (existingPatient.address && existingPatient.address[0]) {
+        const address = existingPatient.address[0];
+        Object.keys(address).forEach(prop => {
+          switch (prop) {
+            case 'id':
+              patientUuidMap['preferredAddressUuid'] = address[prop];
+              break;
+            case 'city':
+              initialAddressFieldValues['cityVillage'] = address[prop];
+              break;
+            case 'state':
+              initialAddressFieldValues['stateProvince'] = address[prop];
+              break;
+            case 'district':
+              initialAddressFieldValues['countyDistrict'] = address[prop];
+              break;
+            case 'extension':
+              address[prop].forEach(ext => {
+                ext.extension.forEach(extension => {
+                  initialAddressFieldValues[extension.url.split('#')[1]] = extension.valueString;
+                });
+              });
+              break;
+            default:
+              if (prop === 'country' || prop === 'postalCode') {
+                initialAddressFieldValues[prop] = address[prop];
+              }
+          }
+        });
+      }
+    }
+  }, [existingPatient]);
+
   const getNames = (values: FormValues) => {
     const names = [
       {
+        uuid: patientUuidMap['preferredNameUuid'],
         preferred: true,
         givenName: values.givenName,
         middleName: values.middleName,
         familyName: values.familyName,
       },
     ];
-    values.addNameInLocalLanguage &&
+    if (values.addNameInLocalLanguage) {
       names.push({
+        uuid: patientUuidMap['additionalNameUuid'],
         preferred: false,
         givenName: values.additionalGivenName,
         middleName: values.additionalMiddleName,
         familyName: values.additionalFamilyName,
       });
+    } else if (patientUuidMap['additionalNameUuid']) {
```

Comment:
Can you explain this section? Are you deleting additional names?

Reply:
If we reach this point, it means the user intended to remove the additional names by probably unchecking the "Add name" field, see:  

![2020-09-16 12-36-19 2020-09-16 12_39_08](https://user-images.githubusercontent.com/26084581/93320460-c7e89200-f819-11ea-8600-3a3b5e716ba3.gif)

Reply:
hmm I'm not sure thats what you want. I mean unchecking the *Add name* doesn't mean delete additional names. A user might still uncheck it if they don't need to add a new name to the already existing names. Maybe in the edit screen we change that label? If it's going to be used to delete additional names then it should read `Add name`.

Reply:
I have no strong preference for sure on this, but what I think is to be intuitive as much as possible to the end user.

If a user clicks the "Edit Patient" button and they find that they have additional names that the want to get rid of, I feel it makes sense if they just unchecked the "Add Name" check button since by it's nature it makes the additional name disappear when checked. So the user unchecks it, and the name disappears, I'm sure they want to think that the name was removed.

>I mean unchecking the Add name doesn't mean delete additional names
Btw, it's just one name. Not a bunch of person names. Maybe there was a lot of hard coding here.

@rbuisson what's your thought on this?